### PR TITLE
Check if tools are installed.

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Add checks for docker, protoc, and golangci-lint tool
- Fixes https://github.com/guacsec/guac/issues/472

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -103,3 +103,22 @@ start-service:
 .PHONY: stop-service
 stop-service:
 	docker compose down
+
+# Check that docker is installed.
+.PHONY: check-docker-tool-check
+check-docker-tool-check:
+	@docker --version > /dev/null
```
Comment (lumjjb) [inline Makefile:110]: can we make it print a note that tool X needs to be installed if it fails?

**Review by lumjjb — COMMENTED** [review]
Minor change requested to add a statement print about why it failed, otherwise LGTM!

File: Makefile (hunk 1 above)
Comment (naveensrinivasan) [reply Makefile]: Took care of it. Thanks

**Review by lumjjb — APPROVED** [review]
lgtm thanks!

**Review by pxp928 — APPROVED** [review]
LGTM

**Pull request merged**